snuba/datasets/entities/sessions.py

```diff
@@ -79,6 +91,127 @@ def plus_columns(
     ]
 )
 
+quantity = Column(None, None, "quantity")
+seq = Column(None, None, "seq")
+status = Column(None, None, "status")
+session_id = Column(None, None, "session_id")
+distinct_id = Column(None, None, "distinct_id")
+duration = Column(None, None, "duration")
+lit_nil = Literal(None, NIL_UUID)
+
+
+def eq(col: Column, lit: Literal) -> FunctionCall:
+    return FunctionCall(None, "equals", (col, lit))
+
+
+def neq(col: Column, lit: Literal) -> FunctionCall:
+    return FunctionCall(None, "notEquals", (col, lit))
+
+
+def _and(ex1: Expression, ex2: Expression) -> FunctionCall:
+    return FunctionCall(None, "and", (ex1, ex2))
+
+
+def _in(col: Column, args: Iterator[Literal]) -> FunctionCall:
+    return FunctionCall(None, "in", (status, FunctionCall(None, "tuple", tuple(args))))
+
+
+has_errors = FunctionCall(
+    None, "greater", (Column(None, None, "errors"), Literal(None, 0))
+)
+did_not_nil = neq(distinct_id, lit_nil)
+duration_condition = _and(
+    neq(duration, Literal(None, MAX_UINT32)), eq(status, Literal(None, 1))
+)
+
+# These here are basically the same statements as the matview query
+sessions_raw_translators = TranslationMappers(
+    columns=[
+        ColumnToCurriedFunction(
+            None,
+            "duration_quantiles",
+            FunctionCall(None, "quantilesIf", quantiles),
+            (duration, duration_condition),
+        ),
+        ColumnToFunction(None, "duration_avg", "avgIf", (duration, duration_condition)),
+        ColumnToFunction(
+            None, "sessions", "sumIf", (quantity, eq(seq, Literal(None, 0))),
+        ),
```

Comment:
You are not including the old data with quantity not populated. We noticed the default is not reliable on distributed tables and we cannot guarantee that people will have quantity populated by the time this starts being used. I think you need to do a countIf + sumIf based on the value of quantity.

Reply:
This was actually intentional. We want to have strict limits on how much into the past you can use this, so this would be restricted to "new" data anyway. So I would rather opt for simplicity here.

Reply:
Ok,
though if you want to limit the range of the query, that should be implemented as a validator for the entity (which is something we do not support yet, but easy to add). @evanh, this is an interesting use case. This entity should be limiting the time range of the query only depending on the granularity of the query itself. I think it is a good use case for custom entity validators.

Reply:
Something that could be added into the `required_conditions` of an Entity once that concept is merged. It wouldn't be difficult to expand that concept to be programmatic instead of statically coded.